src/borg/archive.py

```diff
@@ -1142,7 +1164,7 @@ def item_is_hardlink_master(item):
                     'source' not in item and
                     not matcher.match(item.path))
 
-        for item in archive.iter_items():
+        for item in archive.iter_items(filter=lambda item: 'checkpoint' not in item):
```

Comment:
In that case it may be preferable to have something like a `pre_predicate` on the `Archive` that is ANDed with the iter_items predicate. That would make it easy to control this with a cmdline switch via e.g. `with_archive`.